commit/merkleroot/types.go

```diff
@@ -0,0 +1,156 @@
+		MerkleRoots:        make(map[cciptypes.ChainSelector][]cciptypes.MerkleRootChain),
+		OnRampMaxSeqNums:   make(map[cciptypes.ChainSelector][]cciptypes.SeqNum),
+		OffRampNextSeqNums: make(map[cciptypes.ChainSelector][]cciptypes.SeqNum),
+		FChain:             make(map[cciptypes.ChainSelector][]int),
+	}
+
+	for _, ao := range aos {
+		obs := ao.Observation
+		// MerkleRoots
+		for _, merkleRoot := range obs.MerkleRoots {
+			aggObs.MerkleRoots[merkleRoot.ChainSel] =
+				append(aggObs.MerkleRoots[merkleRoot.ChainSel], merkleRoot)
+		}
+
+		// OnRampMaxSeqNums
+		for _, seqNumChain := range obs.OnRampMaxSeqNums {
+			aggObs.OnRampMaxSeqNums[seqNumChain.ChainSel] =
+				append(aggObs.OnRampMaxSeqNums[seqNumChain.ChainSel], seqNumChain.SeqNum)
+		}
+
+		// OffRampNextSeqNums
+		for _, seqNumChain := range obs.OffRampNextSeqNums {
+			aggObs.OffRampNextSeqNums[seqNumChain.ChainSel] =
+				append(aggObs.OffRampNextSeqNums[seqNumChain.ChainSel], seqNumChain.SeqNum)
+		}
+
+		// FChain
+		for chainSel, f := range obs.FChain {
+			aggObs.FChain[chainSel] = append(aggObs.FChain[chainSel], f)
+		}
+	}
+
+	return aggObs
+}
+
+// ConsensusObservation holds the consensus values for all chains across all observations in a round
+type ConsensusObservation struct {
+	// A map from chain selectors to each chain's consensus merkle root
+	MerkleRoots map[cciptypes.ChainSelector]cciptypes.MerkleRootChain
+
+	// A map from chain selectors to each chain's consensus OnRamp max sequence number
+	OnRampMaxSeqNums map[cciptypes.ChainSelector]cciptypes.SeqNum
+
+	// A map from chain selectors to each chain's consensus OffRamp next sequence number
+	OffRampNextSeqNums map[cciptypes.ChainSelector]cciptypes.SeqNum
+
+	// A map from chain selectors to each chain's consensus f (failure tolerance)
+	FChain map[cciptypes.ChainSelector]int
+}
+
+type OutcomeType int
+
+const (
+	ReportIntervalsSelected OutcomeType = iota + 1
+	ReportGenerated
+	ReportEmpty
+	ReportInFlight
+	ReportTransmitted
+	ReportTransmissionFailed
+)
+
+type Outcome struct {
+	OutcomeType                     OutcomeType                 `json:"outcomeType"`
+	RangesSelectedForReport         []plugintypes.ChainRange    `json:"rangesSelectedForReport"`
+	RootsToReport                   []cciptypes.MerkleRootChain `json:"rootsToReport"`
+	OffRampNextSeqNums              []plugintypes.SeqNumChain   `json:"offRampNextSeqNums"`
+	ReportTransmissionCheckAttempts uint                        `json:"reportTransmissionCheckAttempts"`
+}
+
+// Sort all fields of the given Outcome
+func (o *Outcome) Sort() {
+	sort.Slice(o.RangesSelectedForReport, func(i, j int) bool {
+		return o.RangesSelectedForReport[i].ChainSel < o.RangesSelectedForReport[j].ChainSel
+	})
+	sort.Slice(o.RootsToReport, func(i, j int) bool {
+		return o.RootsToReport[i].ChainSel < o.RootsToReport[j].ChainSel
+	})
+	sort.Slice(o.OffRampNextSeqNums, func(i, j int) bool {
+		return o.OffRampNextSeqNums[i].ChainSel < o.OffRampNextSeqNums[j].ChainSel
```

Comment:
Are we guaranteed to only have a single item per `ChainSel` in each of these slices? I'm pretty sure the answer is yes. If it isn't then this logic is wrong and it should consider `ChainSel` and the other value.